odoo/addons/base/models/res_users.py

```diff
@@ -1665,11 +1665,18 @@ def init(self):
             key varchar not null,
             create_date timestamp without time zone DEFAULT (now() at time zone 'utc')
         );
-        CREATE INDEX IF NOT EXISTS res_users_apikeys_user_id_index_idx ON {table} (user_id, index);
+        CREATE INDEX IF NOT EXISTS {table}_user_id_index_idx ON {table} (user_id, index);
```

Comment:
It would probably be a good idea to create the index name in Python with a bit more checking / safety: if the table name is very long, we might be dropping some of the later bits, and possibly trying to create an index with the exact same name as the table.

Reply:
Also while at it something which should probably be fixed: I don't actually know what happens with `create table`, but even if it doesn't need to do anything `CREATE INDEX IF NOT EXIST` will first take a `SHARE` lock (which completely blocks any table update) *then* check if it needs to create the index. This might be an issue on upgrades if the system is not taken offline during the upgrade.

So might be a good idea to perform the CREATE INDEX in a separate `execute` with a Python-level check beforehand.